rhino/src/main/java/org/mozilla/javascript/EmbeddedSlotMap.java

```diff
@@ -257,9 +256,8 @@ private static void copyTable(Slot[] oldSlots, Slot[] newSlots) {
      */
     private static void addKnownAbsentSlot(Slot[] addSlots, Slot slot) {
         final int insertPos = getSlotIndex(addSlots.length, slot.indexOrHash);
-        Slot old = addSlots[insertPos];
+        slot.next = addSlots[insertPos];
```

Comment:
Both changes in embeddedSlotMap save ~ 5 bytecode instructions: https://godbolt.org/z/1EjhsWs7E